todo/todo.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014 Google Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to writing, software distributed
+// under the License is distributed on a "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package todo provides a type and several functions for interacting
+// with todos backed by the Appengine Datastore. Todo entities are stored
+// as descendants of the same parent, such that the lookups can be made
+// strongly consistent.
+package todo
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	"google.golang.org/appengine/datastore"
+)
+
+// getTodoGroupKey returns a key representing the parent of all Todo entities.
+func getTodoGroupKey(c context.Context) *datastore.Key {
+	return datastore.NewKey(c, "TodoGroup", "", 1, nil)
+}
+
+type Todo struct {
+	ID    int64  `datastore:"-" json:"id"` // Unique identifier
```

Comment:
Yeah, IIRC I wrote it this way because TodoMVC's API expects a todo's key to just be a number: https://github.com/tastejs/todomvc-api/blob/master/todos.apib